src/components/navigation/right-nav.tsx

```diff
@@ -13,19 +13,58 @@ interface IProps extends IBaseProps {
   isGhostUser: boolean;
 }
 
+interface IState {
+  myWorkLoadAllowed: boolean;
+  classWorkLoadAllowed: boolean;
+  classLogsLoadAllowed: boolean;
```

Comment:
These could also be properties of a single object, which could be indexed by tab name. Not a big deal either way.

Reply:
I like it.  Easier to follow.  Fixed.